meson.build

```diff
@@ -242,10 +242,29 @@ if build_daemon
     warning('Polkit is disabled, the daemon will allow ALL client actions')
   endif
   udevdir = get_option('udevdir')
-  if udevdir == ''
+  if udevdir == '' and host_machine.system() == 'linux'
     udev = dependency('udev')
     udevdir = udev.get_pkgconfig_variable('udevdir')
   endif
+  # Check if D-bus is enabled
+  fs = import('fs')
+  dbus_enabled = false
+  machine_id_paths = [
+    '/var/local/lib/dbus/machine-id',
+    '/var/local/etc/machine-id"',
+    '/etc/machine-id',
+    '/var/lib/dbus/machine-id',
+    '/var/db/dbus/machine-id',
+  ]
+  foreach arg: machine_id_paths
+    if fs.is_file(arg)
+      dbus_enabled = true
+      break
+    endif
+  endforeach
+  if not dbus_enabled and not host_machine.system() == 'windows'
+    error('D-Bus is required for the daemon to function properly')
+  endif
```

Comment:
This isn't really checking if dbus is enabled, this is just checking to location of the machine-id file. I wonder if we're going to all this trouble in meson.build we should just set something in config.h and then use that rather than searching in fwupd again.

Reply:
@hughsie I've push that idea as separate commit.

Reply:
@hughsie I implemented this idea in the last commit.